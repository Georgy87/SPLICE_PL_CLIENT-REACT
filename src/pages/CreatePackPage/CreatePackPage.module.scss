.root {
	width: 100vw;
}

.stepOne {
	display: flex;
	flex-direction: column;
}

.stepTwo {
	display: flex;
}

.bottomPanel {
	display: flex;
	justify-content: center;
	padding: 2rem;
	button {
		cursor: pointer;
		&:first-child {
			margin-right: 3px;
		}
		&:last-child {
			margin-left: 3px;
		}
	}
}

.userPacks {
	display: flex;
	width: 100%;
	margin-top: 20px;
}
