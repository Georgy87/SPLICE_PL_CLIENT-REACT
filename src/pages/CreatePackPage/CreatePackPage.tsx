import React, { useState } from 'react';

import { fetchCreatePack } from '../../store/slices/packSlice';
import { useAsyncAction } from '../../hooks/useAsyncAction';
import { StepLayout } from '../../layouts/StepLayout/StepLayout';
import { PackInfoUpload } from '../../components/PackInfoUpload/PackInfoUpload';
import { FileUpload } from '../../components/FileUpload/FileUpload';

import { ButtonLayout } from '../../layouts/ButtonLayout/ButtonLayout';

import styles from '../../styles/pagesStyles/CreatePackPage.module.scss';
import { IconLayout } from '../../layouts/IconLayout/IconLayout';

export const CreatePackPage = () => {
	const [activeStep, setActiveStep] = useState(0);
	const [info, setInfo] = useState({});
	const [picture, setPicture] = useState(null);
	const [audio, setAudio] = useState(null);

	const createTrack = useAsyncAction<any, any>(fetchCreatePack);

	const next = () => {
		if (activeStep != 3) {
			setActiveStep((prev) => prev + 1);
		}

		if (activeStep === 2) {
			console.log(picture, audio);
		}
	};

	const back = () => {
		setActiveStep((prev) => prev - 1);
	};

	return (
		<div>
			<StepLayout activeStep={activeStep}>
				{activeStep === 0 && (
					<div className={styles.stepOne}>
						<PackInfoUpload setInfo={setInfo} />
					</div>
				)}
				{activeStep === 1 && (
					<div className={styles.stepTwo}>
						<FileUpload setFile={setPicture} accept='image/*'>
							<ButtonLayout typeStyle='download'>
								<IconLayout iconName='upload' />
										UPLOAD IMAGE
							</ButtonLayout>
						</FileUpload>
					</div>
				)}
				{activeStep === 2 && (
					<div>
						<FileUpload setFile={setAudio} accept='audio/*'>
							<button>Загрузить аудио</button>
						</FileUpload>
					</div>
				)}
				{activeStep === 3 && (
					<div className={styles.send}>
						<div
							onClick={() =>
								createTrack({ info, picture, audio })
							}
						>
							Отправить
						</div>
					</div>
				)}
			</StepLayout>
			<div className={styles.bottomPanel}>
				<ButtonLayout
					typeStyle={activeStep === 0 ? 'blue-disabled' : 'blue'}
					disabled={activeStep === 0}
					onClicked={back}
				>
					Назад
				</ButtonLayout>
				<ButtonLayout typeStyle='blue' onClicked={next}>
					Далее
				</ButtonLayout>
			</div>
		</div>
	);
};
