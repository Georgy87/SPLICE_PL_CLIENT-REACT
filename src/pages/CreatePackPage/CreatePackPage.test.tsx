import { render } from '@testing-library/react';

import { renderWithRouter } from '../../tests/helpers/RenderWithRouter';
import { CreatePackPage } from './CreatePackPage';

describe('CreatePackPage component', () => {
	test('CreatePackPage snapshot', async () => {
		const createPackPage = render(renderWithRouter(<CreatePackPage />, '/profile/create'));
		expect(createPackPage).toMatchSnapshot();
	});
});
