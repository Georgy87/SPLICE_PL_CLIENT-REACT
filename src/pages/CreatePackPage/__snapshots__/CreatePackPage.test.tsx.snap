// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePackPage component CreatePackPage snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="root"
        data-testid="create-pack-page"
      >
        <div
          class="stepContainer"
        >
          <div
            class="stepNames"
          >
            <div
              class="step"
            >
              <div
                class="step"
              >
                <div
                  class="step stepRed"
                >
                  pack information
                  <div
                    class="line"
                  />
                </div>
              </div>
            </div>
            <div
              class="step"
            >
              <div
                class="step"
              >
                <div
                  class="step stepBlue"
                >
                  pack cover
                  <div
                    class="line"
                  />
                </div>
              </div>
            </div>
            <div
              class="step"
            >
              <div
                class="step"
              >
                <div
                  class="step stepBlue"
                >
                  pack audio
                  <div
                    class="line"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="stepUploadInfo"
          >
            <div
              class="stepOne"
            >
              <form
                class="packUploadWrapper"
              >
                <label>
                  GENRE
                </label>
                <input
                  name="genre"
                  type="text"
                />
                <p />
                <label>
                  AUTHOR NAME
                </label>
                <input
                  name="authorName"
                  type="text"
                />
                <p />
                <label>
                  DESCRIPTION
                </label>
                <textarea
                  name="description"
                />
                <p />
                <div
                  class="platformBtn"
                >
                  <button
                    class="platformBtn black"
                  >
                    SAVE INFO
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
        <div
          class="bottomPanel"
        >
          <div
            class="platformBtn"
          >
            <button
              class="platformBtn blueDisabled"
              disabled=""
            >
              PREV
            </button>
          </div>
          <div
            class="platformBtn"
          >
            <button
              class="platformBtn blue"
            >
              NEXT
            </button>
          </div>
        </div>
      </div>
      <div
        class="root"
        data-testid="create-pack-page"
      >
        <div
          class="stepContainer"
        >
          <div
            class="stepNames"
          >
            <div
              class="step"
            >
              <div
                class="step"
              >
                <div
                  class="step stepRed"
                >
                  pack information
                  <div
                    class="line"
                  />
                </div>
              </div>
            </div>
            <div
              class="step"
            >
              <div
                class="step"
              >
                <div
                  class="step stepBlue"
                >
                  pack cover
                  <div
                    class="line"
                  />
                </div>
              </div>
            </div>
            <div
              class="step"
            >
              <div
                class="step"
              >
                <div
                  class="step stepBlue"
                >
                  pack audio
                  <div
                    class="line"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="stepUploadInfo"
          >
            <div
              class="stepOne"
            >
              <form
                class="packUploadWrapper"
              >
                <label>
                  GENRE
                </label>
                <input
                  name="genre"
                  type="text"
                />
                <p />
                <label>
                  AUTHOR NAME
                </label>
                <input
                  name="authorName"
                  type="text"
                />
                <p />
                <label>
                  DESCRIPTION
                </label>
                <textarea
                  name="description"
                />
                <p />
                <div
                  class="platformBtn"
                >
                  <button
                    class="platformBtn black"
                  >
                    SAVE INFO
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
        <div
          class="bottomPanel"
        >
          <div
            class="platformBtn"
          >
            <button
              class="platformBtn blueDisabled"
              disabled=""
            >
              PREV
            </button>
          </div>
          <div
            class="platformBtn"
          >
            <button
              class="platformBtn blue"
            >
              NEXT
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="root"
      data-testid="create-pack-page"
    >
      <div
        class="stepContainer"
      >
        <div
          class="stepNames"
        >
          <div
            class="step"
          >
            <div
              class="step"
            >
              <div
                class="step stepRed"
              >
                pack information
                <div
                  class="line"
                />
              </div>
            </div>
          </div>
          <div
            class="step"
          >
            <div
              class="step"
            >
              <div
                class="step stepBlue"
              >
                pack cover
                <div
                  class="line"
                />
              </div>
            </div>
          </div>
          <div
            class="step"
          >
            <div
              class="step"
            >
              <div
                class="step stepBlue"
              >
                pack audio
                <div
                  class="line"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="stepUploadInfo"
        >
          <div
            class="stepOne"
          >
            <form
              class="packUploadWrapper"
            >
              <label>
                GENRE
              </label>
              <input
                name="genre"
                type="text"
              />
              <p />
              <label>
                AUTHOR NAME
              </label>
              <input
                name="authorName"
                type="text"
              />
              <p />
              <label>
                DESCRIPTION
              </label>
              <textarea
                name="description"
              />
              <p />
              <div
                class="platformBtn"
              >
                <button
                  class="platformBtn black"
                >
                  SAVE INFO
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        class="bottomPanel"
      >
        <div
          class="platformBtn"
        >
          <button
            class="platformBtn blueDisabled"
            disabled=""
          >
            PREV
          </button>
        </div>
        <div
          class="platformBtn"
        >
          <button
            class="platformBtn blue"
          >
            NEXT
          </button>
        </div>
      </div>
    </div>
    <div
      class="root"
      data-testid="create-pack-page"
    >
      <div
        class="stepContainer"
      >
        <div
          class="stepNames"
        >
          <div
            class="step"
          >
            <div
              class="step"
            >
              <div
                class="step stepRed"
              >
                pack information
                <div
                  class="line"
                />
              </div>
            </div>
          </div>
          <div
            class="step"
          >
            <div
              class="step"
            >
              <div
                class="step stepBlue"
              >
                pack cover
                <div
                  class="line"
                />
              </div>
            </div>
          </div>
          <div
            class="step"
          >
            <div
              class="step"
            >
              <div
                class="step stepBlue"
              >
                pack audio
                <div
                  class="line"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="stepUploadInfo"
        >
          <div
            class="stepOne"
          >
            <form
              class="packUploadWrapper"
            >
              <label>
                GENRE
              </label>
              <input
                name="genre"
                type="text"
              />
              <p />
              <label>
                AUTHOR NAME
              </label>
              <input
                name="authorName"
                type="text"
              />
              <p />
              <label>
                DESCRIPTION
              </label>
              <textarea
                name="description"
              />
              <p />
              <div
                class="platformBtn"
              >
                <button
                  class="platformBtn black"
                >
                  SAVE INFO
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        class="bottomPanel"
      >
        <div
          class="platformBtn"
        >
          <button
            class="platformBtn blueDisabled"
            disabled=""
          >
            PREV
          </button>
        </div>
        <div
          class="platformBtn"
        >
          <button
            class="platformBtn blue"
          >
            NEXT
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
